### PR TITLE
WMS GetLegendGraphic fix for #15055

### PR DESCRIPTION
**Pull request by rduivenvoorde**
See http://hub.qgis.org/issues/15055

When creating the legend image url, this tests for available queryparams
in a case-insensitive way...

To test and about the problem, see issue.

I tried to use Q_FOREACH but was not able to make it work. Feel free to make it better.

### CODE REVIEW

File: src/providers/wms/qgswmsprovider.cpp [hunk 1]
```diff
@@ -2990,19 +2990,24 @@ QUrl QgsWmsProvider::getLegendGraphicFullURL( double scale, const QgsRectangle&
 
   QUrl url( lurl );
 
-  if ( !url.hasQueryItem( "SERVICE" ) )
+  // query names/keys are NOT case-sensitive, so make an uppercase list for proper comparison
+  QList<QString> qnames;
+  for(int i = 0; i < url.queryItems().size(); i++) {
+    qnames << url.queryItems().at(i).first.toUpper();
```
Comment (sbrunner) [inline src/providers/wms/qgswmsprovider.cpp]: The style check will complain about it..., you should add some space e.g.: `(i)` => `( i )`

Comment (jef-n) [inline src/providers/wms/qgswmsprovider.cpp]: better run ``scripts/prepare-commit.sh``...

**Comment by sbrunner** [comment]
For my that's looks good, thanks :-)

**Comment by rduivenvoorde** [comment]
Ok, I ran the script. Let me know if this is ok like this.
I can also retract the pull request and do this in one commit?

**Comment by jef-n** [comment]
you can ``git rebase -i HEAD~2`` squash the two commits and then ``git push -f``

**Comment by rduivenvoorde** [comment]
Thanks Juergen, this was my first 'squash' :-)

File: src/providers/wms/qgswmsprovider.cpp [hunk 2]
```diff
@@ -2990,19 +2990,25 @@ QUrl QgsWmsProvider::getLegendGraphicFullURL( double scale, const QgsRectangle&
 
   QUrl url( lurl );
 
-  if ( !url.hasQueryItem( "SERVICE" ) )
+  // query names/keys are NOT case-sensitive, so make an uppercase list for proper comparison
+  QList<QString> qnames;
```
Comment (nyalldawson) [inline src/providers/wms/qgswmsprovider.cpp]: Very minor change, but lists of strings should use QStringList.

**Comment by nyalldawson** [comment]
If you rebase again :P, try renaming the commit to include the exact string "fix #15055" .

**Comment by nyalldawson** [comment]
Great work @rduivenvoorde !

**Comment by rduivenvoorde** [comment]
@nyalldawson OK. Done. Thanks.

**Comment by rduivenvoorde** [comment]
@nyalldawson @jef-n anything else to do? If not can somebody plz pull in?

**Pull request merged**